server/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -708,16 +740,240 @@ public static <T extends Throwable> T readStackTrace(T throwable, StreamInput in
     /**
      * Serializes the given exceptions stacktrace elements as well as it's suppressed exceptions to the given output stream.
      */
-    public static <T extends Throwable> T writeStackTraces(T throwable, StreamOutput out, Writer<Throwable> exceptionWriter)
-        throws IOException {
+    public static void writeStackTraces(Throwable throwable, StreamOutput out, Writer<Throwable> exceptionWriter) throws IOException {
         out.writeArray((o, v) -> {
             o.writeString(v.getClassName());
             o.writeOptionalString(v.getFileName());
             o.writeString(v.getMethodName());
             o.writeVInt(v.getLineNumber());
         }, throwable.getStackTrace());
         out.writeArray(exceptionWriter, throwable.getSuppressed());
-        return throwable;
+    }
+
+    /**
+     * Writes the specified {@code throwable} to {@link StreamOutput} {@code output}.
+     */
+    public static void writeException(Throwable throwable, StreamOutput output) throws IOException {
+        writeException(throwable, output, () -> {});
```

Comment:
For the nested exception callback to be as useful as before, I think the default here needs to have the same assert. The reason we have an assert at all is so we can fail explicitly when in tests.

However, I'm also on the fence about whether such an assertion is actually useful. We would only get it when running unit/integ tests, but not when running rest tests. We can discuss as a followup though.